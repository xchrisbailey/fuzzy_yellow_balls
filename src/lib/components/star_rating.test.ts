import { render, fireEvent } from '@testing-library/svelte';
import StarRating from './star_rating.svelte';

describe('StarRating component', () => {
	it('should render correctly', () => {
		const { getByLabelText } = render(StarRating, { name: 'rating', max: 5, step: 1, value: 0 });

		expect(getByLabelText('Rating')).toBeInTheDocument();
	});

	it('should update value when range input changes', async () => {
		const { getByLabelText } = render(StarRating, { name: 'rating', max: 5, step: 1, value: 0 });
		const input = <HTMLInputElement>getByLabelText('Rating');

		await fireEvent.input(input, { target: { value: 3 } });

		expect(input.value).toBe('3');
	});
});
