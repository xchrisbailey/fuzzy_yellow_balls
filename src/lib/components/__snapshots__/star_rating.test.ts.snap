// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StarRating > should render a star rating 1`] = `
<body>
  <div>
    <label
      class="label s-eyYaLZR3NLvp"
    >
      <span
        class="s-eyYaLZR3NLvp"
      >
        Test
      </span>
       
      <input
        class="rating s-eyYaLZR3NLvp"
        max="5"
        name="test"
        step="1"
        type="range"
      />
    </label>
    <!--&lt;Star_rating&gt;-->
  </div>
</body>
`;
